sdk/core/azure-core/src/http/http.cpp

```diff
@@ -153,3 +155,13 @@ void Azure::Core::Http::Details::InsertHeaderWithValidation(
   // insert (override if duplicated)
   headers[headerName] = headerValue;
 }
+
+std::string Azure::Core::Http::Internal::RangeToString(Range range)
+{
+  std::string rangeString = "bytes=" + std::to_string(range.Offset) + "-";
+  if (range.Length != std::numeric_limits<decltype(range.Length)>::max())
```

Comment:
Why use `2^63 - 1` as the "until the end" sentinel value when `-1` (or any negative length, really) would be easier to handle?

Reply:
The length needs to be nullable, it's optional, and we should use the fact it isn't set as the sentinel to leave it out of the string.